presto-main/src/main/java/com/facebook/presto/cost/JoinStatsRule.java

```diff
@@ -313,7 +314,7 @@ private PlanNodeStatsEstimate calculateJoinComplementStats(
 
         // TODO: use range methods when they have defined (and consistent) semantics
         double leftNDV = leftColumnStats.getDistinctValuesCount();
-        double matchingRightNDV = rightColumnStats.getDistinctValuesCount() * unmatchedJoinComplementNdvsCoefficient;
+        double matchingRightNDV = roundToDouble(rightColumnStats.getDistinctValuesCount() * unmatchedJoinComplementNdvsCoefficient);
```

Comment:
copying from https://github.com/prestodb/presto/pull/11856#discussion_r232169603

Why would you want to round NDV estimate?
1.5 is a perfect average expected value. Rounding it to 2 introduces 33% error.

Reply:
i see your response now https://github.com/prestodb/presto/pull/11856#discussion_r232493865 (i was able to see my question but not your response on GH)

> There are no 1.5 distinct values in real world. There are either 2 or 1. Same as for the rows count. It makes it way easier to reason about the cost model when integer values are integers. In real world we cannot have 1.5 rows returned.
> 
> I had to introduce that round to make leftNDV <= matchingRightNDV check work in some of the corner cases covered by the unit tests (as value returned by SymbolStatsEstimate#getDistinctValuesCount is always rounded now).


I do not agree. 
I know the final result will be either 1 NDV or 2 (or some other integer).
However, we are not trying to "guess" the final result (in guessing, betting on a possible answer would make a lot of sense). Instead, we are trying to approximate the final result, and here, approximating with something that is not a whole integer is perfectly valid.

Reply:
I understand the expected value concept. I'm just not sure if the increased accuracy worth of always keeping in mind that NDV and Row count values could be between 0 and 1.